bin/crew

```diff
@@ -1283,15 +1283,18 @@ def install
   # Debug code...
   puts 'before json change'
   system "/usr/bin/jq --arg key ruby -e '.installed_packages[] | select(.name == \$key )' /usr/local/etc/crew/device.json"
+
+  # Clean out package from json before adding, especially if package may be in CREW_ESSENTIAL_PACKAGES.
+  @device[:installed_packages].delete_if { |entry| entry['name'] == @pkg.name }
```

Comment:
Why is this necessary?  Doesn't it just get added back in again?  Wouldn't a better approach be to check if the package exists and if not, add it?

Reply:
A second one gets added in if it isn't first removed.

We need
`If the name is in the array
  remove it
then add name
...
`

Reply:
A second one is added only because you're not checking if it exists beforehand.

Reply:
Adjusted code to remove the entry if it is in `CREW_ESSENTIAL_PACKAGES`.